docs2/site/docs/getting-started/transport.md

```diff
@@ -0,0 +1,89 @@
+    - `complete` - Bidirectional. This message indicate that the GraphQL subscription has completed.
+
+    #### graphql-transport-ws
+    - `GQL_CONNECTION_INIT` - Client -> Server. Client sends this message to initialize the connection.
+    - `GQL_CONNECTION_ACK` - Server -> Client. Server sends this message to acknowledge the connection.
+    - `GQL_CONNECTION_ERROR` - Server -> Client. Server sends this message to indicate that an error occurred while establishing the connection.
+    - `GQL_CONNECTION_TERMINATE` - Client -> Server. Client sends this message to terminate the connection.
+    - `GQL_START` - Client -> Server. Client sends this message to start a GraphQL operation.
+    - `GQL_DATA` - Server -> Client. Server sends this message to send the GraphQL execution result.
+    - `GQL_ERROR` - Server -> Client. Server sends this message to indicate that an error occurred during the GraphQL operation.
+    - `GQL_COMPLETE` - Server -> Client. Server sends this message to indicate that the GraphQL operation is complete.
+    - `GQL_STOP` - Client -> Server. Client sends this message to stop a running GraphQL operation.
+- Payload - (Optional, object): The payload of the message. The payload is only used for *subscribe*, *GQL_DATA*, *GQL_ERROR*, and *GQL_COMPLETE* messages.
+
+**Note:** s mentioned in [Apollo Client](https://www.apollographql.com/docs/react/data/subscriptions/#websocket-setup) documentation, the *graphql-transport-ws* is a legacy protocol and should not be used in new applications. The *GraphQL-WS* protocol is the recommended protocol to use.
+
+## IGraphQLSerializer & IGraphQLTextSerializer
+
+Serialize and deserialize object hierarchies to/from a `stream` and `string` respectively. Should include special support for `ExecutionResult` , `Inputs` and transport-specific classes as necessary. Dotnet-GraphQL provides `Newtonsoft.Json` and `System.Text.Json` as default implementations of this interfaces. 
+
+
+Simple example of `System.Text.Json` serializer in a asp.net controller is shown below:
+
+*Program.cs*
+```csharp
+builder.Services.AddGraphQL(options =>
+    {
+        options.AddSystemTextJson();
+        options.AddSchema<FooSchema>();
+    });
+```
+
+*Controller.cs*
+```csharp
+public GraphQLController(ISchema schema, IDocumentExecuter executer, IGraphQLTextSerializer serializer)
+    {
+        _schema = schema;
+        _executer = executer;
+        _serializer = serializer;
+    }
+
+[HttpPost]
+public async Task<IActionResult> Post([FromBody] GraphQLRequest query)
+    {
+        var result = await _executer.ExecuteAsync(options =>
+            {
+                options.Schema = _schema;
+                options.Query = query.Query;
+            }).ConfigureAwait(false);
+            if (result.Errors?.Count > 0)
+            {
+                return BadRequest();
+            }
+
+            return Ok(_serializer.Serialize(result));
```

Comment:
```suggestion
        Response.StatusCode = (int)(result.Executed ? HttpStatusCode.OK : HttpStatusCode.BadRequest);
        return Content(_serializer.Serialize(result), "application/graphql-response+json", System.Text.Encoding.UTF8);
```

Reply:
This represents the recommended behavior after 1/1/2025, excluding optional support for `Accept: application/json`.  Notice that when `Executed` is `true` the status code is 200, even if a failure occurs during execution.  Only for validation errors does 400 get returned.

The current recommended behavior prior to 1/1/2025 is to use `application/json` (unless requested otherwise via the `Accept` header) and always return 200, as it may be difficult to discern GraphQL failures from proxy server failures.

For more details see the draft HTTP over GraphQL spec.

Reply:
Note that GraphQL.NET Server currently adopts the 1/1/2025 behavior in general.  It does also support `Accept: application/json`, however the response status code does not vary depending on the response content type.  See `GraphQLHttpMiddlewareOptions.ValidationErrorsReturnBadRequest`.